### ISSUE TITLE
[Accessibility]: No description for branch pills

### ISSUE THREAD

**Issue by eseliger** [sourcegraph/sourcegraph#35011]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34089

### Problem description

<img width="202" alt="Screenshot 2022-05-05 at 21 13 46@2x" src="https://user-images.githubusercontent.com/19534377/167008855-2bce82a2-6f7c-457f-9e76-ff023c83f2b9.png">
these pills here are just read aloud by the screen reader without any context. They should likely get some sort of aria label.

### Expected behavior

It's clear to the listener what the two words and the arrow refer to.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
@eseliger Any objections to assigning this to GitStart to work on?

**Comment by eseliger** [comment]
nope, go for it!
